code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -111,6 +112,7 @@
 		if(T && istype(radio) && istype(radio.keyslot))
 			radio.keyslot.forceMove(T)
 			radio.keyslot = null
+	UnregisterSignal(src, COMSIG_LIGHT_EATER_ACT)
```

Comment:
Signals get automatically unregistered when the object is deleted already.

Reply:
I know. But this is the standard format.

Reply:
Where is this standard taken from?

Reply:
Done. More overhead, but done.